README.adoc

```diff
@@ -1,3 +1,62 @@
 
+This service takes care of sending email notification for different Notification Type.
+
+=== Contributes:
+
+==== Project Structure:
+
+This project is build around Notification Type in center.
+
+To register new notification type and to see list of exisitng notification type check in [main.go](https://github.com/fabric8-services/fabric8-notification/blob/master/main.go)
+
+===== Introducing New Notification Type:
+
+====== Register:
+
+New Notification Type can be registered as below,
+
+```
+resolvers.Register("analytics.notify.cve", collector.ConfiguredVars(config, collector.NewCVEResolver(authClient, witClient)), nil)
+```
+
+First arg, Notification Type name (string), here it is "analytics.notify.cve".  You need to come up with unique Notification Type name depending on what kind of New Notificatoin you want to support.  Later value of this will be used to get registered Resolver and Validator in NotifyController.Send() 
+Second arg, ReceiverResolver, it is used to find out list of users to which email should be sent when Notificatin arrives for given Notification Type, more on this later.
+Third arg, ParamValidator, it is used to validate parameter received in Notification payload.
+
+====== Resolver:
+
+ReceiverResolver is responsible to return list of email to which notification sent for given Notification Type.  Resolver will get `ID` attribute as input.
+
+The value of `ID` attibute should be decided such that the related emails can be fetched.
+For ex, to find notification email list for `workitem.create` type, one would like to get `Workitem.ID` for newly created workitem.
+
+You can also decide to set `ID` with value of `codebase_url` for `analytics.notify.cve` type as with url one can find users who have registred for CVE notification on given Codebase URL.
+
+Resolver will generally get initialized with 'WIT Client' and 'Auth Client' which will be used to query data based on `ID` to get notification email list.
+
+====== Template:
+
+This is about email template which include email body, subject, headers etc.  All email templates are maintained in [template](https://github.com/fabric8-services/fabric8-notification/tree/master/template) folder.  For each Notification Type there is one folder with its name.
+
+body.html - mandatory file which should have email body.  It uses golang template [link](https://golang.org/pkg/text/template/).
+subject.txt - mandatory file which should have subject like for email.  It also uses golnag template.
+
+==== Build Project:
+
+===== Prerequisites:
+
+One should have go, git, mercurial, make installed on the system.
+
+===== make commands:
+
+Run `make all` to build the project.
+
+Run `make preview` to run [preview/main.go](https://github.com/fabric8-services/fabric8-notification/blob/master/preview/main.go), more on this later.
```

Comment:
link misformatted. and maybe also remove the `more on this later`, because this will be a living document as opposed to a one time blog.